### PR TITLE
Bump supported and recommended k8s versions for kops 1.19

### PR DESCRIPTION
**Pull request by johngmyers**
/milestone v1.19
/hold
(for 1.18 to branch)

### CODE REVIEW

File: docs/tutorial/upgrading-kubernetes.md [hunk 1]
```diff
@@ -118,10 +118,10 @@ Restart the instances with `kops rolling-update cluster --yes`.
 ```
 > kubectl get nodes -owide
 NAME                        STATUS    AGE       VERSION   EXTERNAL-IP     OS-IMAGE                             KERNEL-VERSION
-master-us-central1-a-8fcc   Ready     26m       v1.7.7    35.194.56.129   Container-Optimized OS from Google   4.4.35+
-nodes-9cml                  Ready     16m       v1.7.7    35.193.12.73    Ubuntu 16.04.3 LTS                   4.10.0-35-generic
-nodes-km98                  Ready     10m       v1.7.7    35.194.25.144   Ubuntu 16.04.3 LTS                   4.10.0-35-generic
-nodes-wbb2                  Ready     2m        v1.7.7    35.188.177.16   Ubuntu 16.04.3 LTS                   4.10.0-35-generic
+master-us-central1-a-8fcc   Ready     26m       v1.17.7   35.194.56.129   Container-Optimized OS from Google   4.4.35+
+nodes-9cml                  Ready     16m       v1.17.7   35.193.12.73    Ubuntu 16.04.3 LTS                   4.10.0-35-generic
```
Comment (MoShitrit) [inline docs/tutorial/upgrading-kubernetes.md:122]: Should we also update the image versions of Ubuntu to `18.04 / 20.04`? Or is 16.04 still the default for now?

Comment (johngmyers) [reply docs/tutorial/upgrading-kubernetes.md:122]: I didn't bother to research that.

File: pkg/model/components/docker.go [hunk 2]
```diff
@@ -73,17 +73,13 @@ func (b *DockerOptionsBuilder) BuildOptions(o interface{}) error {
...
-		docker.Storage = fi.String("overlay2,overlay,aufs")
-	} else {
-		docker.Storage = fi.String("overlay,aufs")
-	}
+	// TODO(justinsb): figure out whether to use overlay2 on AWS jessie:
+	// The ContainerOS image now has docker configured to use overlay2 out-of-the-box
+	// and it is an error to specify the flag twice.
+	// But Jessie (still our default AWS image) isn't recommended by docker with overlay2
+	// (though that may be a kernel issue, and we run a custom kernel on our default image)
+	// But we still need to worry about users running generic AMIs (e.g. stock jessie)
```
[the remark this answers is not in the document]
Comment (johngmyers) [reply pkg/model/components/docker.go:81]: Not sure what to do with the ContainerOS reference.

[the remark this answers is not in the document]
Comment (hakman) [reply pkg/model/components/docker.go:81]: 2 years is quite a lot to be relevant anymore. Maybe just keep the COS part of the comment.

File: pkg/model/components/docker.go [hunk 3]
```diff
@@ -73,17 +73,9 @@ func (b *DockerOptionsBuilder) BuildOptions(o interface{}) error {
...
-		// But Jessie (still our default AWS image) isn't recommended by docker with overlay2
-		// (though that may be a kernel issue, and we run a custom kernel on our default image)
-		// But we still need to worry about users running generic AMIs (e.g. stock jessie)
-		docker.Storage = fi.String("overlay2,overlay,aufs")
-	} else {
-		docker.Storage = fi.String("overlay,aufs")
-	}
+	// TODO(justinsb): The ContainerOS image now has docker configured to use overlay2 out-of-the-box
+	// and it is an error to specify the flag twice.
+	docker.Storage = fi.String("overlay2,overlay,aufs")
```
Comment (johngmyers) [inline pkg/model/components/docker.go:78]: @hakman Are overlay and aufs still relevant? (It would be a separate PR if not.)

Comment (hakman) [reply pkg/model/components/docker.go:78]: All supported distros should have support for overlay2, so I think you are right and the rest could be removed.

Comment (hakman) [reply pkg/model/components/docker.go:78]: Removing this completely is also a possibility, as Docker already has a pretty good list of priorities:
https://github.com/docker/docker-ce/blob/v19.03.11/components/engine/daemon/graphdriver/driver_linux.go#L51

File: pkg/model/components/etcdmanager/options.go [hunk 4]
```diff
@@ -57,12 +57,7 @@ func (b *EtcdManagerOptionsBuilder) BuildOptions(o interface{}) error {
 		}
 
 		if etcdCluster.Version == "" {
-			if b.IsKubernetesGTE("1.11") {
-				etcdCluster.Version = "3.2.18"
-			} else {
-				// Preserve existing default etcd version
-				etcdCluster.Version = "2.2.1"
```
Comment (hakman) [inline pkg/model/components/etcdmanager/options.go:64]: Remove `2.2.1` also from `supportedEtcdVersions`.

File: pkg/model/components/etcd.go [hunk 5]
```diff
@@ -83,10 +83,8 @@ func (b *EtcdOptionsBuilder) BuildOptions(o interface{}) error {
 				c.Version = DefaultEtcd3Version_1_14
 			} else if b.IsKubernetesGTE("1.13") {
 				c.Version = DefaultEtcd3Version_1_13
-			} else if b.IsKubernetesGTE("1.11") {
-				c.Version = DefaultEtcd3Version_1_11
 			} else {
-				c.Version = DefaultEtcd2Version
```
Comment (hakman) [inline pkg/model/components/etcd.go:89]: Remove also const `DefaultEtcd2Version`.

File: pkg/model/components/etcdmanager/options.go (hunk 4 above)
Comment (johngmyers) [reply pkg/model/components/etcdmanager/options.go:64]: Cannot. Used as default for legacy provider with k8s 1.11 and 1.12

File: pkg/model/components/etcd.go (hunk 5 above)
Comment (johngmyers) [reply pkg/model/components/etcd.go:89]: Cannot. Several remaining references.

File: pkg/model/components/etcdmanager/options.go (hunk 4 above)
Comment (hakman) [reply pkg/model/components/etcdmanager/options.go:64]: Ah.. Thanks

File: pkg/model/components/docker.go (hunk 3 above)
Comment (johngmyers) [reply pkg/model/components/docker.go:78]: I look forward to your future PR.

Comment (hakman) [reply pkg/model/components/docker.go:78]: 👍

**Comment by hakman** [comment]
Would be nice to get this merged. Feels like there's a few rebases every day.
/lgtm
